pkg/controllers/state/cluster.go

```diff
@@ -669,6 +669,9 @@ func (c *Cluster) updateNodePoolResources(oldNode, newNode *StateNode) {
 	if _, ok := c.nodePoolResources[newNodePoolName]; !ok && newNodePoolName != "" {
 		c.nodePoolResources[newNodePoolName] = corev1.ResourceList{}
 	}
+	if _, ok := c.nodePoolResources[oldNodePoolName]; !ok && oldNodePoolName != "" {
```

Comment:
I don't think this is right -- this basically means that we would subtract from the oldNodePool -- realistically, this should never occur so I think something else may be going on here